pages/css-grid-superposition--qcj5jo/css-grid-superposition--qcj5jo.md

```diff
@@ -0,0 +1,279 @@
+We won't cover layout shift happening on first page load.
+
+## Let's get some context
+
+I work for [Clever Cloud](https://www.clever-cloud.com/en/), an IT automation platform.
+Our clients push their code and we handle the rest for them: build, deploy, host, scale, maintain, recover...
+When they need to configure their applications, databases and other services, they use our web UI: "the console".
+I spend most of my time on this project and it looks like this:
+
+![Console overview of a Node.js application](./console-overview.png)
+
+What you see in the screenshot above is the overview of my own site, a Node.js application hosted on our platform.
+On the right, you can see a pie chart which represents the repartition of HTTP response codes returned by the application in the last 24 hours.
+
+![UI Component displaying a chart](./cc-tile-status-codes-chart.png)
+
+<!-- TODO -->
+When I worked on this component, I imagined our users would implicitly understand the chart is related to the current application.
+I decided to move some details like the exact number of requests in tooltips accessible via mouse and touch.
+The problem was, how do I explain the following details:
+
+* The chart only displays data from the last 24 hours.
+* The legend items can be clicked to show/hide the different status code categories. Thank you [Chart.js](https://www.chartjs.org/)!
+
+The available space was too small to add a title at the top or bottom of the chart so to help our users figure this out, I added an information button ℹ️ in the top right corner.
+When they click on it, a short text is displayed in place of the chart like this:
+
+![UI Component displaying some help](./cc-tile-status-codes-help.png)
+
+When they click on the cross icon, we hide the short text and display the chart again.
+Implementing this toggle behaviour was not that hard.
+Let's take a simplified version of the template:
+
+```html
+<div class="chart-component">
+  <div class="title">
+    HTTP response codes <button>toggle chart/info</button>
+  </div>
+  <div class="chart">
+    <!-- chart here -->
+  </div>
+  <div class="info">
+    <!-- short text here -->
+  </div>
+</div>
+```
+
+With this HTML, we could simply toggle the component state between `.chart` and `.info` when the button is clicked and hide the appropriate panel with `display: none`.
+
+## Layout shift problems
+
+This approach has one big problem, it is subject to **layout shift**.
+
+> What is that?
+
+When I work on such components, I tend to follow what I consider a good practice: don't set a fixed height on the component, let it adapt to its own content.
+This is even more important in our situation because we have no way to know how much space we need to display the short text (without scrollbars).
+The minimum required height will depend on:
+
+* The unknown width of the component, which depends on usage context
+* The length of the short text, which depends on the translation (english or french)
```

Comment:
```suggestion
* The length of the short text, which depends on the language (english or french)
```

We have the habit of saying "translation" in this context, but I'm pretty sure most people don't.